test/differencer/rulesProcessor.test.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+import { NodeDiff } from "../../src/diffTool/jsonDiff";
+import { applyRules } from "../../src/diffTool/rulesProcessor";
+import * as chai from "chai";
+import path from "path";
+import fs from "fs-extra";
+import tmp from "tmp";
+import sinon from "sinon";
+import { ramlToolLogger } from "../../src";
+import chaiAsPromised from "chai-as-promised";
+
+/* eslint-disable @typescript-eslint/no-use-before-define*/
+
+const expect = chai.expect;
+const rulesPath = path.join(__dirname, "..", "..", "diffRules", "rules.json");
+let loggerSpy;
+before(() => {
+  chai.should();
+  chai.use(chaiAsPromised);
+  loggerSpy = sinon.spy(ramlToolLogger, "info");
+});
+after(() => {
+  sinon.reset();
+});
+
+describe("Test that rules application exits with relevant message when there are no differences", () => {
+  const noDiffsMsg = "No differences to apply the rules";
+  it("exits when the differences are undefined", async () => {
+    const diffs = undefined;
+    await applyRules(undefined, rulesPath);
+    expect(diffs).to.equal(diffs);
+    sinon.assert.calledWith(loggerSpy, noDiffsMsg);
+  });
+  it("exits when there are empty differences", async () => {
+    const diffs = [];
+    await applyRules(diffs, rulesPath);
+    expect(diffs).to.deep.equal(diffs);
```

Comment:
```suggestion
    expect(diffs).to.deep.equal([]);
```
An object will always deep equal itself, so you need a second object to compare.